src/Core__Option.res

```diff
@@ -22,73 +22,85 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. */
 
+let some = value => Some(value)
```

Comment:
Why do we need this?

Reply:
They're useful to pipe into, because data constructors can't be used as functions, and so the alternative is something like `->(x => Some(x))`, which is a bit awkward to both type and read in the middle of a sequence of pipes.

Reply:
Also useful to pass to `map` functions, though less so than the above.

Reply:
Could you do an example of when `Some` doesn't work in piping? I'm having trouble reproducing it. Check this out for example: https://rescript-lang.org/try?version=v10.1.2&code=DYUwLgBA5uByCuBbCBeCA7JqB8GsGoIBGABgChRIAPVaOJACiICYBmASgFpsBlAe0QhuAIRDAwAOgDyABzABLPugmIAhjIYA3HBG2EiXbKPHS5i5TDABRKum79BZIA

Reply:
Hmm, yeah, apparently this has worked since at least 8.3. I had no idea! Not sure if this is something I have from earlier versions, or I've just assumed it didn't work because it doesn't work as a function in other contexts.

Also tested if `Some(_)` produces a function that can be passed to `map`, and a bit disappointed to find that it does not. So there's still a little bit of a case for it, but certainly much less.

Reply:
Might be because it doesn't work with `|>`? But because `a->Some` is just a syntax transform to `Some(a)`, it works.

Reply:
Could be! I've jumped around between Reason, OCaml and ReScript a bit, so that might have gotten me confused.

Reply:
I say we skip this for now when the use case is fairly limited. Can always bring it in later if there's a compelling case.

Reply:
Removed